app/src/org/commcare/activities/StandardHomeActivity.java

```diff
@@ -246,5 +263,60 @@ public boolean usesSubmissionProgressBar() {
     public void refreshUI() {
         uiController.refreshView();
     }
-    
+
+    @Override
+    protected void onDestroy() {
+        appUpdateController.unregister();
+        super.onDestroy();
+    }
+
+    private void handleAppUpdate() {
+        AppUpdateState state = appUpdateController.getStatus();
+        switch (state) {
+            case UNAVAILABLE:
+                break;
+            case AVAILABLE:
+                StandardAlertDialog alertDialog = StandardAlertDialog.getBasicAlertDialog(this,
+                        Localization.get("in.app.update.available.title"),
+                        Localization.get("in.app.update.available.detail"),
+                        null);
+                alertDialog.setPositiveButton(Localization.get("in.app.update.dialog.update"), (dialog, which) -> {
+                    appUpdateController.startUpdate(StandardHomeActivity.this);
+                    dismissAlertDialog();
+                });
+                alertDialog.setNegativeButton(Localization.get("in.app.update.dialog.no.thanks"), (dialog, which) -> {
+                    appUpdateController.skipVersion();
```

Comment:
I think it's a good idea to avoid asking users to update CommCare to a specific version if they deny installing the update

Reply:
can we have a counter instead, for eg. if they deny 3 times for a particular version then don't show it to them. User might just be busy at times and can deny because of that.

Reply:
Actually I think it's fine to remove the prompt once user denies it for a particular update 2-3 times though can we give a menu option which only become visible in case an app update is available so that user still has some way to trigger the app update workflow.